### DISCUSSION TITLE
Vacuum followed by mop for some rooms

### DISCUSSION THREAD

**Discussion by wigster** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#583] [Q&A]
Hi,

I am using your card with a Roborock S7 (as of yesterday with the HACS Roborock integration, before with the Xiaomi one). I mainly use the vacuum by picking rooms through the card and letting it go in custom mode, so that in each room I get different fan/mop modes as set in the phone app.

I would like to make one change to the config -- for some rooms, I want to have the S7 first vacuum without mop and then go back and vacuum/mop again before moving to the new room. which would again play out the custom sequence for that next room. I guess I should have a separate script set up for each segment which would be the default action for that room if it is selected. 

This is impossible to set in the app -- can I do it with your card somehow?

**Comment by PiotrMachowski** [comment]
You can create a parametrized script that will receive a list of rooms to clean from the card and clean every room according to your preferences. After creating such script you can create an additional entry in `map_modes` (with custom `service_call_schema`) that will execute it.
